dspy/clients/lm.py

```diff
@@ -88,7 +88,16 @@ def __init__(
         model_pattern = re.match(r"^(?:o[1345]|gpt-5)(?:-(?:mini|nano))?", model_family)
 
         if model_pattern:
-            if max_tokens < 16000 or temperature != 1.0:
+            if temperature == 0 and max_tokens == 4000:
```

Comment:
Per our offline discussion, let's proceed with `temperature=None` and `max_tokens=None` as the default, which aligns with litellm defaults.